build_bins.sh

```diff
@@ -45,14 +45,15 @@ apt-get -y update
 # Install tools needed for packaging
 apt-get -y install git rubygems make pkg-config pbuilder python-mock python-configobj python-support cdbs python-all-dev python-stdeb libmysqlclient-dev libldap2-dev ruby-dev gcc patch rake ruby1.9.3 ruby1.9.1-dev python-pip python-setuptools dpkg-dev apt-utils haveged libtool autoconf automake autotools-dev unzip rsync autogen
 
-if [[ -z `gem list --local cabin | grep cabin | cut -f1 -d" "` ]]; then
-  gem install cabin --no-ri --no-rdoc -v 0.7.2
-fi
-
+# Install json gem first to avoid a too-new version being pulled in by other gems.
 if [[ -z `gem list --local json | grep json | cut -f1 -d" "` ]]; then
   gem install json --no-ri --no-rdoc -v 1.8.3
 fi
 
+if [[ -z `gem list --local cabin | grep cabin | cut -f1 -d" "` ]]; then
+  gem install cabin --no-ri --no-rdoc -v 0.7.2
+fi
+
```

Comment:
Yeah, that would be better, but I don't see it happening as part of the shell script builds.